### ISSUE TITLE
Munki 2.8.1 breaks on macOS 10.6

### ISSUE THREAD

**Issue by fseesink** [munki/munki#672]
Yeah yeah, I know, ancient gear, so not expecting much here.  But thought I'd mention it all the same.

I still have a few Apple XServes that are functioning perfectly running macOS 10.6.8 which, until relatively recently, worked fine with Munki.  But I noticed now with 2.8.1 that if I attempt to run the following, I get the output shown:

```
$ munkiimport --version
Traceback (most recent call last):
  File "/usr/local/munki/munkiimport", line 67, in <module>
    globals=NetFS, scan_classes=False)
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/PyObjC/objc/_bridgesupport.py", line 210, in initFrameworkWrapper
    frameworkName, "PyObjC.bridgesupport"):
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/pkg_resources.py", line 824, in resource_exists
    return get_provider(package_or_requirement).has_resource(resource_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/pkg_resources.py", line 160, in get_provider
    __import__(moduleOrReq)
ImportError: No module named NetFS
```

Now this may have broken with Munki 2.8.0, I'm not sure.  I notice that Firefox on the machines in question is at 48.0.2, which was released on 24 Aug 2016.  So it broke in the past month or so.

Anyway, just thought I'd throw that out there.  You still list Munki as supporting 10.6 and up.  Thought you might want to know this.

**Comment by gregneagle** [comment]
Thanks for the report -- I don't do much testing of Munki on 10.6.x any longer, but I do try to keep it running on Snow Leopard.
This is almost certainly the new share mounting code added to address changes in Sierra. It should be possible to conditionally import/use this code on newer OSes and fall back to the older share mounting code on older OSes.
Since this affects munkiimport only, the workaround in the meantime would be to do your munkiimporting on a machine running a newer version of OS X.

**Comment by fseesink** [comment]
Oh, sorry, I should've been more clear.  I just posted the output from munkiimport as that's the one I typically run just to see what version is installed.  But I noticed that the Managed Software Center wasn't working either.  Firing that up and clicking on [Updates] resulted in the message:

> Update check failed
> There is a configuration problem with the managed software installer. Could not start the process. Contact your systems administrator.
> 

Going back to Terminal, here's the output from running that:

```
$ managedsoftwareupdate --version
Traceback (most recent call last):
  File "/usr/local/munki/managedsoftwareupdate", line 59, in <module>
    from munkilib import updatecheck
  File "/usr/local/munki/munkilib/updatecheck.py", line 35, in <module>
    import appleupdates
  File "/usr/local/munki/munkilib/appleupdates.py", line 37, in <module>
    import fetch
  File "/usr/local/munki/munkilib/fetch.py", line 38, in <module>
    from gurl import Gurl
  File "/usr/local/munki/munkilib/gurl.py", line 37, in <module>
    from CFNetwork import kCFNetworkProxiesHTTPSEnable, kCFNetworkProxiesHTTPEnable
ImportError: cannot import name kCFNetworkProxiesHTTPSEnable

```

**Comment by gregneagle** [comment]
Thanks. That's a different and obviously more important issue. It's unlikely I'll have a fix until next week.

**Comment by gregneagle** [comment]
First issue should be fixed by https://github.com/munki/munki/commit/23672ca187e37541cf48c4166fa8c388076109eb

Second issue should be fixed by https://github.com/munki/munki/commit/0269b1a89cee45694274609b74842ea420c9f761

Please build from source and test, or test against this:
https://munkibuilds.org/2.8.1.2850/

**Comment by fseesink** [comment]
Tested using https://github.com/munki/munki/releases/tag/v2.8.2RC1

Yep, seems to be working again.  Thanks very much.  I realize these units are old.  Ok, truth be told, they're still pretty darn fast, but the OS on them is old since Apple didn't make 64-bit drivers for the video card, which seems to be the lynchpin change when 10.7 came out.  So appreciate being able to maintain them with Munki.  Says something that this software works with so many versions of the OS.

**Comment by gregneagle** [comment]
Resolved with 2.8.2RC2.
